cognite/client/data_classes/data_modeling/typed_instances.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+
+from abc import ABC
+from collections.abc import Iterable
+from datetime import date, datetime
+from typing import TYPE_CHECKING, Any, ClassVar
+
+from typing_extensions import Self
+
+from cognite.client.data_classes._base import CogniteResource
+from cognite.client.data_classes.data_modeling.data_types import (
+    DirectRelationReference,
+)
+from cognite.client.data_classes.data_modeling.ids import ContainerId, EdgeId, NodeId, ViewId
+from cognite.client.data_classes.data_modeling.instances import PropertyValueWrite
+from cognite.client.utils._text import to_camel_case
+
+if TYPE_CHECKING:
+    from cognite.client import CogniteClient
+
+
+class InstanceProperty:
+    """This is a descriptor class for instance properties in a typed class.
+
+    It is used when you have a property that has a different name in the Data Model
+    compared to the name in the Python class.
+    """
+
+    def __init__(self, alias: str | None = None) -> None:
```

Comment:
It's not really an alias, the attribute name is really the alias and this is the actual identifier. Maybe `identifer` or `identifier_override`?
```suggestion
    def __init__(self, identifier: str | None = None) -> None:
```